web/src/components/vis/VisTile.vue

```diff
@@ -11,28 +11,61 @@ export default {
     },
   },
 
+  data() {
+    return {
+      scale: 1,
+    };
+  },
+
   computed: {
-    hasControls() {
-      return !!this.$slots.controls;
+    smallEnough() {
+      return this.scale <= 0.5;
+    },
+    largeEnough() {
+      return this.scale >= 4;
+    },
+    scaleClass() {
+      return `scale${Math.round(this.scale * 100)}`;
     },
   },
 };
 </script>
 
 <template lang="pug">
-v-flex(shrink=1)
+v-flex(shrink=1, :class="scaleClass")
   .white.rounded.relative
     v-toolbar.primary.darken-3.top-rounded(dark, flat, dense)
       v-toolbar-title {{ title }}
       v-spacer
-      v-toolbar-items(v-if="hasControls")
+      v-toolbar-items
+        v-btn(@click="scale *= 0.5", :disabled="smallEnough", icon)
+          v-icon {{ $vuetify.icons.viewCollapse }}
+        v-btn(@click="scale *= 2", :disabled="largeEnough", icon)
+          v-icon {{ $vuetify.icons.viewExpand }}
         slot(name="controls")
     v-progress-linear.ma-0.progress(v-if="loading", indeterminate, height=4)
     v-card.bottom-rounded(flat)
-      slot
+      slot(:scale="scale")
 </template>
 
 <style scoped lang="scss">
+.scale50 {
+  grid-column: span 1;
+  grid-row: span 1;
+}
+.scale100 {
+  grid-column: span 2;
+  grid-row: span 2;
+}
+.scale200 {
+  grid-column: span 4;
+  grid-row: span 4;
+}
+.scale400 {
```

Comment:
prefer `scale1, scale2, scale3` etc.  The actual numbers don't seem to matter, so you could use grid layouts that aren't powers of 2.

Reply:
`span 4` refers to that four cells should be used. `scale200` mean a scaling factor of  200 percent. Thus, you can read it as: When the scaling factor is `200`% the cell should span over four cells in the grid layout

Reply:
I suppose my point is that the math here, while technically correct, isn't necessary.  You've already hard-coded the result of the equation `cell width * 200%` within the CSS class.

All you really have is a finite set of constant classes you can apply, and you iterate through them by generating their string keys mathematically. 

I think you could reduce the complexity by having something like below where you just have buttons that increment a counter up and down.  A bonus here would be that your min and max go away, and you iterate from 0 to `sizeClasses.length - 1`

```
<script>
  ...
  data() { return { sizeClasses = [ 'class1', 'class2', 'class3', ...] }},
<template>
   ...
   <element :class="sizeClasses[index]"
</template>
```

Reply:
Another consideration is that the math ties us unnecessarily to this exponential scale.  We might find that a Fibonacci scale is more appropriate, and if the class name generation weren't mathematical, all you have to do is change a few CSS properties.

Reply:
+1 for simplifying the logic here.